DataSource integration

Nashatyrev opened the pull request:
Some efforts to check how the `DataSource` framework (from EthereumJ) may fit to the Pantheon architecture. 
The code compiles but is not intended to work. Some classes are just stubs

Nashatyrev committed:
Temporarily comment test due to removed classes

Nashatyrev committed:
Partial DataSource implementation and partial core refactoring

Nashatyrev committed:
Make WorldStateStorage simpler. Updater functionality looks superfluous here: if at some point one would need processing parallel updates this could be achieved by creating several WorldStateStorage instances (with independent WriteCaches)

mkalinin commented on ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/worldstate/DefaultMutableWorldState.java:
```diff
@@ -48,6 +48,7 @@
   private final Map<Address, BytesValue> updatedAccountCode = new HashMap<>();
   private final WorldStateStorage worldStateStorage;
 
+
```
Looks like this is an accidental empty line.

Nashatyrev committed:
Some code polish, add javadoc

Nashatyrev committed:
return original method name flush() instead of commit()
commit() is more about transactions, atomicity and isolation
all these properties are not generally applies to this method

Nashatyrev committed:
Add draft interfaces for transactions support

Nashatyrev committed:
Comment test temporarily

Nashatyrev committed:
Make kvstore pure DataSource implementation

Nashatyrev committed:
Refactor KeyValueStoragePrefixedKeyBlockchainStorage for DataSource usage, get rid of KeyValueStorage dependency

Nashatyrev committed:
Get rid of core -> kvstore module dependency
